features/request/request.feature

```diff
@@ -0,0 +1,55 @@
+Feature: Formatting requests for sending to the Email Service API
+
+  Scenario: Checking site status without validating the email service ip address
+    Given I am using a baseUri of "https://api.example.com/"
+      And I have indicated not to validate the email service ip
+    When I call getSiteStatus
+    Then the method should be "GET"
+      And the url should be "https://api.example.com/site/status"
+
+  Scenario: Checking site status with trusted email-service
+   Given I am using a trusted baseUri
+     And I have indicated that I want the email service ip to be validated
+    When I call getSiteStatus
+    Then the method should be "GET"
+    And the url should be "https://trusted_host.org/site/status"
+
+  Scenario: Checking client with untrusted email-service
+    Given I am using an untrusted baseUri
+    When I create the emailServiceClient
+    Then an UntrustedIp exception will be thrown
+
+  Scenario: Checking client with a single trusted ip block value
+    Given I have not indicated whether the email service ip should be validated
+      And I am using a single value for a trusted ip block
+    When I create the emailServiceClient
+    Then an InvalidArgument exception will be thrown
+
+  Scenario: Checking client with the assert_valid_ip not given and the trusted_ip_ranges value missing
```

Comment:
Since having hard-to-read tests can lead to its own set of problems, I can update this Behat test as written here after this PR is merged. No reason to hold things up for that refactor, though.